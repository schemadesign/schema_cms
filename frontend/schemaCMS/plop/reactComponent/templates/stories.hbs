import React from 'react';
import { storiesOf } from '@storybook/react';

import { {{ pascalCase name }} } from './{{ camelCase name }}.component';

export const defaultProps = {};

storiesOf('{{ pascalCase name }}', module).add('Default', () => <{{ pascalCase name }} {...defaultProps} />);
