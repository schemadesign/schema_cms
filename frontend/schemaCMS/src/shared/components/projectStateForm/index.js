export { ProjectStateForm } from './projectStateForm.component';
