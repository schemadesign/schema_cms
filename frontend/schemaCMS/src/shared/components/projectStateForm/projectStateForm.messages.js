/* eslint-disable max-len */
import { defineMessages } from 'react-intl';
import {
  PROJECT_STATE_SOURCE_URL,
  PROJECT_STATE_DATA_SOURCE,
  PROJECT_STATE_DESCRIPTION,
  PROJECT_STATE_NAME,
  PROJECT_STATE_AUTHOR,
  PROJECT_STATE_CREATED,
  PROJECT_STATE_IS_PUBLIC,
} from '../../../modules/projectState/projectState.constants';

export default defineMessages({
  [PROJECT_STATE_DATA_SOURCE]: {
    id: `shared.components.projectStateForm.${[PROJECT_STATE_DATA_SOURCE]}`,
    defaultMessage: 'Dataset',
  },
  dataSourcePlaceholder: {
    id: 'shared.components.projectStateForm.dataSourcePlaceholder',
    defaultMessage: 'Select a dataset',
  },
  [PROJECT_STATE_NAME]: {
    id: `shared.components.projectStateForm.${[PROJECT_STATE_NAME]}`,
    defaultMessage: 'Name',
  },
  [PROJECT_STATE_DESCRIPTION]: {
    id: `shared.components.projectStateForm.${[PROJECT_STATE_DESCRIPTION]}`,
    defaultMessage: 'Description',
  },
  [PROJECT_STATE_SOURCE_URL]: {
    id: `shared.components.projectStateForm.${[PROJECT_STATE_SOURCE_URL]}`,
    defaultMessage: 'Source URL',
  },
  [PROJECT_STATE_AUTHOR]: {
    id: `shared.components.projectStateForm.${[PROJECT_STATE_AUTHOR]}`,
    defaultMessage: 'Author',
  },
  [PROJECT_STATE_CREATED]: {
    id: `shared.components.projectStateForm.${[PROJECT_STATE_CREATED]}`,
    defaultMessage: 'Created',
  },
  [PROJECT_STATE_IS_PUBLIC]: {
    id: `shared.components.projectStateForm.${[PROJECT_STATE_IS_PUBLIC]}`,
    defaultMessage: 'Make Public',
  },
});
