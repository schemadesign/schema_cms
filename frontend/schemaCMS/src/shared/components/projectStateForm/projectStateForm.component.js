import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { always, find, path, pipe, prop, propEq } from 'ramda';
import { Form } from 'schemaUI';
import dayjs from 'dayjs';

import { Container } from './projectStateForm.styles';
import { TextInput } from '../form/inputs/textInput';
import messages from './projectStateForm.messages';
import { Select } from '../form/select';
import {
  PROJECT_STATE_AUTHOR,
  PROJECT_STATE_CREATED,
  PROJECT_STATE_DATA_SOURCE,
  PROJECT_STATE_DESCRIPTION,
  PROJECT_STATE_IS_PUBLIC,
  PROJECT_STATE_NAME,
  PROJECT_STATE_SOURCE_URL,
} from '../../../modules/projectState/projectState.constants';
import { renderWhenTrue, renderWhenTrueOtherwise } from '../../utils/rendering';

const { Switch } = Form;

export class ProjectStateForm extends PureComponent {
  static propTypes = {
    intl: PropTypes.object.isRequired,
    handleChange: PropTypes.func.isRequired,
    setFieldValue: PropTypes.func.isRequired,
    values: PropTypes.object.isRequired,
    state: PropTypes.object,
    dataSources: PropTypes.array.isRequired,
  };

  static defaultProps = {
    state: {},
  };

  getStatusOptions = () =>
    this.props.dataSources.map(({ name, id }) => ({
      value: id,
      label: name,
    }));

  handleSelectStatus = ({ value }) => {
    this.props.setFieldValue(PROJECT_STATE_DATA_SOURCE, value);
  };

  renderInput = (value, name) =>
    renderWhenTrue(
      always(
        <TextInput
          value={value}
          onChange={this.props.handleChange}
          name={name}
          label={this.props.intl.formatMessage(messages[name])}
          fullWidth
          disabled
          {...this.props}
        />
      )
    )(!!value);

  renderSelect = (value, stateValue, name) =>
    renderWhenTrueOtherwise(
      always(
        <TextInput
          value={pipe(path(['props', 'dataSources']), find(propEq('id', value)), prop('name'))(this)}
          onChange={this.props.handleChange}
          name={name}
          label={this.props.intl.formatMessage(messages[name])}
          fullWidth
          disabled
          {...this.props}
        />
      ),
      always(
        <Select
          label={this.props.intl.formatMessage(messages[name])}
          name={name}
          value={value}
          options={this.getStatusOptions()}
          onSelect={this.handleSelectStatus}
          placeholder={this.props.intl.formatMessage(messages.dataSourcePlaceholder)}
        />
      )
    )(!!stateValue);

  render() {
    const { handleChange, intl, values, state } = this.props;

    return (
      <Container>
        {this.renderSelect(
          values[PROJECT_STATE_DATA_SOURCE],
          state[PROJECT_STATE_DATA_SOURCE],
          PROJECT_STATE_DATA_SOURCE
        )}
        <TextInput
          value={values[PROJECT_STATE_NAME]}
          onChange={handleChange}
          name={PROJECT_STATE_NAME}
          label={intl.formatMessage(messages[PROJECT_STATE_NAME])}
          fullWidth
          isEdit
          {...this.props}
        />
        <TextInput
          value={values[PROJECT_STATE_DESCRIPTION]}
          onChange={handleChange}
          name={PROJECT_STATE_DESCRIPTION}
          label={intl.formatMessage(messages[PROJECT_STATE_DESCRIPTION])}
          fullWidth
          isEdit
          multiline
          {...this.props}
        />
        <TextInput
          value={values[PROJECT_STATE_SOURCE_URL]}
          onChange={handleChange}
          name={PROJECT_STATE_SOURCE_URL}
          label={intl.formatMessage(messages[PROJECT_STATE_SOURCE_URL])}
          fullWidth
          isEdit
          {...this.props}
        />
        {this.renderInput(state[PROJECT_STATE_AUTHOR], PROJECT_STATE_AUTHOR)}
        {this.renderInput(dayjs(state[PROJECT_STATE_CREATED]).format('DD/MM/YYYY'), PROJECT_STATE_CREATED)}
        <Switch
          value={values[PROJECT_STATE_IS_PUBLIC]}
          id={PROJECT_STATE_IS_PUBLIC}
          onChange={handleChange}
          label={intl.formatMessage(messages[PROJECT_STATE_IS_PUBLIC])}
        />
      </Container>
    );
  }
}
