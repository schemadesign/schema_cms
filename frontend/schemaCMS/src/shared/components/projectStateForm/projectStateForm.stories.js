import React from 'react';
import { storiesOf } from '@storybook/react';

import { ProjectStateForm } from './projectStateForm.component';

const defaultProps = {};

storiesOf('ProjectStateForm', module).add('Default', () => <ProjectStateForm {...defaultProps} />);
