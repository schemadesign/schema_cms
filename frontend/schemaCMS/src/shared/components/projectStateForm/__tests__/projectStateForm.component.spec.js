import React from 'react';
import { shallow } from 'enzyme';

import { ProjectStateForm } from '../projectStateForm.component';
import { defaultProps } from '../projectStateForm.stories';

describe('ProjectStateForm: Component', () => {
  const component = props => <ProjectStateForm {...defaultProps} {...props} />;

  const render = (props = {}) => shallow(component(props));

  it('should render correctly', () => {
    const wrapper = render();
    global.expect(wrapper).toMatchSnapshot();
  });
});
