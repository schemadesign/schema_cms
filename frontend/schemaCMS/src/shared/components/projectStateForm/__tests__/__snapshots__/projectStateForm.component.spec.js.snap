// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectStateForm: Component should render correctly 1`] = `
<styled.div>
  <Memo()
    label="Dataset"
    name="datasource"
    onSelect={[Function]}
    options={
      Array [
        Object {
          "label": "name",
          "value": "id",
        },
      ]
    }
    placeholder="Select a dataset"
  />
  <TextInput
    autoWidth={false}
    checkOnlyErrors={false}
    dataSources={
      Array [
        Object {
          "id": "id",
          "name": "name",
        },
      ]
    }
    errors={Object {}}
    fullWidth={true}
    handleChange={[Function]}
    intl={
      Object {
        "formatMessage": [Function],
      }
    }
    isEdit={true}
    label="Name"
    name="name"
    onChange={[Function]}
    setFieldValue={[Function]}
    state={
      Object {
        "author": "author",
        "created": "10.12.2019",
        "description": "description",
        "isPublic": false,
        "name": "name",
        "project": 1,
      }
    }
    touched={Object {}}
    value="name"
    values={
      Object {
        "author": "author",
        "created": "10.12.2019",
        "description": "description",
        "isPublic": false,
        "name": "name",
        "project": 1,
      }
    }
  />
  <TextInput
    autoWidth={false}
    checkOnlyErrors={false}
    dataSources={
      Array [
        Object {
          "id": "id",
          "name": "name",
        },
      ]
    }
    errors={Object {}}
    fullWidth={true}
    handleChange={[Function]}
    intl={
      Object {
        "formatMessage": [Function],
      }
    }
    isEdit={true}
    label="Description"
    multiline={true}
    name="description"
    onChange={[Function]}
    setFieldValue={[Function]}
    state={
      Object {
        "author": "author",
        "created": "10.12.2019",
        "description": "description",
        "isPublic": false,
        "name": "name",
        "project": 1,
      }
    }
    touched={Object {}}
    value="description"
    values={
      Object {
        "author": "author",
        "created": "10.12.2019",
        "description": "description",
        "isPublic": false,
        "name": "name",
        "project": 1,
      }
    }
  />
  <TextInput
    autoWidth={false}
    checkOnlyErrors={false}
    dataSources={
      Array [
        Object {
          "id": "id",
          "name": "name",
        },
      ]
    }
    errors={Object {}}
    fullWidth={true}
    handleChange={[Function]}
    intl={
      Object {
        "formatMessage": [Function],
      }
    }
    isEdit={true}
    label="Source URL"
    name="sourceUrl"
    onChange={[Function]}
    setFieldValue={[Function]}
    state={
      Object {
        "author": "author",
        "created": "10.12.2019",
        "description": "description",
        "isPublic": false,
        "name": "name",
        "project": 1,
      }
    }
    touched={Object {}}
    values={
      Object {
        "author": "author",
        "created": "10.12.2019",
        "description": "description",
        "isPublic": false,
        "name": "name",
        "project": 1,
      }
    }
  />
  <TextInput
    autoWidth={false}
    checkOnlyErrors={false}
    dataSources={
      Array [
        Object {
          "id": "id",
          "name": "name",
        },
      ]
    }
    disabled={true}
    errors={Object {}}
    fullWidth={true}
    handleChange={[Function]}
    intl={
      Object {
        "formatMessage": [Function],
      }
    }
    isEdit={false}
    label="Author"
    name="author"
    onChange={[Function]}
    setFieldValue={[Function]}
    state={
      Object {
        "author": "author",
        "created": "10.12.2019",
        "description": "description",
        "isPublic": false,
        "name": "name",
        "project": 1,
      }
    }
    touched={Object {}}
    value="author"
    values={
      Object {
        "author": "author",
        "created": "10.12.2019",
        "description": "description",
        "isPublic": false,
        "name": "name",
        "project": 1,
      }
    }
  />
  <TextInput
    autoWidth={false}
    checkOnlyErrors={false}
    dataSources={
      Array [
        Object {
          "id": "id",
          "name": "name",
        },
      ]
    }
    disabled={true}
    errors={Object {}}
    fullWidth={true}
    handleChange={[Function]}
    intl={
      Object {
        "formatMessage": [Function],
      }
    }
    isEdit={false}
    label="Created"
    name="created"
    onChange={[Function]}
    setFieldValue={[Function]}
    state={
      Object {
        "author": "author",
        "created": "10.12.2019",
        "description": "description",
        "isPublic": false,
        "name": "name",
        "project": 1,
      }
    }
    touched={Object {}}
    value="12/10/2019"
    values={
      Object {
        "author": "author",
        "created": "10.12.2019",
        "description": "description",
        "isPublic": false,
        "name": "name",
        "project": 1,
      }
    }
  />
  <Component
    id="isPublic"
    label="Make Public"
    onChange={[Function]}
    value={false}
  />
</styled.div>
`;
